chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -77,30 +77,19 @@ log "No HA setup found..."
 
 <% if @compute_node %>
 
-log "Upgrading packages needed for nova-compute node"
-
-zypper --non-interactive up openstack-nova-compute
-
 log "Restarting services for Pike"
 
-zypper --non-interactive up <%= @neutron_agent %>
 <% unless @remote_node %>
 systemctl restart <%= @neutron_agent %>
 <% end %>
 
-<% if @l3_agent %>
-zypper --non-interactive up <%= @l3_agent %>
-<% unless @remote_node %>
+<% if @l3_agent && !@remote_node %>
 systemctl restart <%= @l3_agent %>
```

Comment:
Why is this still needed? You don't update the l3-, metadata- and cinder-agents so why would we need to restart?

Reply:
Well I do not update _anything_, not even nova-compute. But those services were down when I got to that point. I guess you want to see more details why this happened.

Reply:
> Well I do not update anything, not even nova-compute.

I know, I just overlooked the fact that the script also still restarts nova-compute (it was hidden by the github UI) 

> But those services were down when I got to that point. I guess you want to see more details why this happened.

Well I guess it's the "crowbar-upgrade.rb" recipe doing it.